libs/core/concurrency/include/hpx/concurrency/deque.hpp

```diff
@@ -29,6 +29,7 @@
 #include <memory>
```

Comment:
Are these changes related to the thread id changes. If yes, how?

Reply:
No, it was a missing #include.